src/resmom_win/win/mom_start.c

```diff
@@ -105,19 +105,19 @@ set_shell(job *pjob, struct passwd *pwdp)
 	 * If we fail to get cmd shell(unlikely), use "cmd.exe" as shell
 	 */
 	if (0 != get_cmd_shell(shell, sizeof(shell)))
-		(void)snprintf(shell, sizeof(shell) - 1, "cmd.exe");
+		strncpy(shell, "cmd.exe", sizeof(shell) - 1);
```

Comment:
I think we might be leaving a chance of having non null turnminated shell string. Please handle this case for all the 3 strncpy changes in this PR

Reply:
We initialize the entire array to null. Since we are copying till sizeof(shell)-1 it will ensure the last element of the array will be null

Reply:
are you sure about that. may be this one is safe as "cmd.exe" is smaller than MAXPATH len but the I'm afraid about the below two strncpy

Reply:
Can you point me where we are initializing the whole buffer ?

Reply:
Will not matter, we are only copying MAX_PATH - 1 in strncpy. Also MAX_PATH is a windows standard.

Reply:
We initialize the whole buffer on line 97.

Reply:
We initialize the whole buffer on line 99.

Reply:
Since the buffer is static my concern was that it only gets initialized the first time only. But 
Yes I think its a false alarm since the "-1" saves the last NULL byte from being overwritten. Now I agree that shell will always be NULL terminated. Thanks

Reply:
Better to not depend on the static variable to never be used again by the next programmer down the line (and overwrite your NULL character at some point). Instead of initializing it at declaration time, just add a line after this:
shell[sizeof(shell) - 1] = '\0' 

It is way faster than setting NULLs on every byte (which you dont need) and fool proof, if somebody overwrites the last NULL

Reply:
I can see that `get_cmd_shell()` is clearing out the entire shell, but I agree that we should have a explicit guard that @subhasisb suggested, for the sake of assurance while reading the code.